### PR TITLE
Enable warnings as errors for Python tests

### PR DESCRIPTION
**Pull request by mroeschke**
As part of https://github.com/rapidsai/build-planning/issues/26, warnings in Python tests will be converted to test failures

`ignore:Unknown pytest.mark.ucx:PytestUnknownMarkWarning` could be removed once https://github.com/rapidsai/raft/pull/2281 is merged cc @jameslamb

### CODE REVIEW

**Comment by jameslamb** [comment]
Thanks for the `@`.

I didn't even think about how this work you're doing would have caught the thing I just happened to notice in logs that led to #2281... nice!!

@rapidsai/raft-python-codeowners #2281 is small and I think non-controversial, it's ready for review.

File: python/raft-dask/pytest.ini [hunk 1]
```diff
@@ -1,4 +1,7 @@
 [pytest]
+filterwarnings =
+    error
+    ignore:Unknown pytest.mark.ucx:PytestUnknownMarkWarning
```
Comment (mroeschke) [inline python/raft-dask/pytest.ini:4]:
```suggestion
```

**Review by dantegd — APPROVED** [review]
lgtm

**Pull request merged**